lib/pacemaker/pcmk_verify.c

```diff
@@ -0,0 +1,172 @@
+        }
+    } else if (xml_stdin) {
+        cib_object = stdin2xml();
+        if (cib_object == NULL) {
+            rc = ENODATA;
+            g_set_error(&error, PCMK__RC_ERROR, rc, "Couldn't parse input from STDIN.");
+        }
+
+    } else {
+        rc = ENODATA;
+        g_set_error(&error, PCMK__RC_ERROR, rc,
+                    "No configuration source specified.  Use --help for usage information.");
+    }
+
+    if (!pcmk__xe_is(cib_object, XML_TAG_CIB)) {
+        rc = EBADMSG;
+        g_set_error(&error, PCMK__RC_ERROR, rc,
+                    "This tool can only check complete configurations (i.e. those starting with <cib>).");
+    }
+
+    if (cib_save != NULL) {
+        write_xml_file(cib_object, cib_save, FALSE);
+    }
+
+    status = pcmk_find_cib_element(cib_object, XML_CIB_TAG_STATUS);
+    if (status == NULL) {
+        create_xml_node(cib_object, XML_CIB_TAG_STATUS);
+    }
+
+    if (pcmk__validate_xml(cib_object, NULL, (xmlRelaxNGValidityErrorFunc) out->err, out) == FALSE) {
+        pcmk__config_err("CIB did not pass schema validation");
+        free_xml(cib_object);
+        cib_object = NULL;
+
+    } else if (cli_config_update(&cib_object, NULL, FALSE) == FALSE) {
+        crm_config_error = TRUE;
+        free_xml(cib_object);
+        cib_object = NULL;
+        out->err(out, "The cluster will NOT be able to use this configuration.\n"
+                 "Please manually update the configuration to conform to the %s syntax.",
+                 xml_latest_schema());
+    }
+
+    scheduler = pe_new_working_set();
+    if (scheduler == NULL) {
+        rc = errno;
+        g_set_error(&error, PCMK__RC_ERROR, rc,
+                    "Could not allocate scheduler data: %s", pcmk_rc_str(rc));
+    }
+    scheduler->priv = out;
+
+    /* Process the configuration to set crm_config_error/crm_config_warning.
+     *
+     * @TODO Some parts of the configuration are unpacked only when needed (for
+     * example, action configuration), so we aren't necessarily checking those.
+     */
+    if (cib_object != NULL) {
+        unsigned long long flags = pcmk_sched_no_counts|pcmk_sched_no_compat;
+
+        if ((status == NULL) && !use_live_cib) {
+            // No status available, so do minimal checks
+            flags |= pcmk_sched_validate_only;
+        }
+        pcmk__schedule_actions(cib_object, flags, scheduler);
+    }
+    pe_free_working_set(scheduler);
+
+    if (crm_config_error) {
+        rc = pcmk_rc_schema_validation;
+
+        if (verbosity > 0 || pcmk__str_eq(output_ty, "xml", pcmk__str_none)) {
+            g_set_error(&error, PCMK__RC_ERROR, rc,
+                        "Errors found during check: config not valid");
+        } else {
+            g_set_error(&error, PCMK__RC_ERROR, rc,
+                        "Errors found during check: config not valid\n-V may provide more details");
+        } 
```

Comment:
I think it's better here if you get rid of the test on the output format and instead just reword the second error message to not have a newline in it.  It could be something like:
`"Errors found during check: config not valid - add -V to increase verbosity"`
Or something along those lines.

You can do the same thing for warnings down below.